### PR TITLE
libxc: Fix compilation after distribution changes

### PR DESCRIPTION
**Pull request by RemiLacroix-IDRIS**
Probably a side effect of a change on Gitlab side.

### CODE REVIEW

**Comment by RemiLacroix-IDRIS** [comment]
I have changed this PR to a draft as it seems that the problem is more complex than I initially realized.

**Comment by RemiLacroix-IDRIS** [comment]
Removed the WIP tag, I have pushed a new commit which fixes all the issues.

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 1]
```diff
@@ -15,24 +15,24 @@ class Libxc(AutotoolsPackage, CudaPackage):
...
+    version("5.1.3", sha256="76b2abd063b692ed7e60fb6dfdf5a54072378710ee91f2b352a4e311d9805e97")
+    version("5.1.2", sha256="ff13eef8184b6c61dac8933ee74fc05967de4a67489581bdc500f1ec63826aae")
+    version("5.1.0", sha256="e8d2b6eb2b46b356a27f0367a7665ff276d7f295da7c734e774ee66f82e56297")
+    version("5.0.0", sha256="6b3be3cf6daf6b3eddf32d4077276eb9169531b42f98c2ca28ac85b9ea408493")
+    version("4.3.4", sha256="2d5878dd69f0fb68c5e97f46426581eed2226d1d86e3080f9aa99af604c65647")
+    version("4.3.2", sha256="3bbe01971d0a43fb63b5c17d922388a39a3f0ae3bd37ae5f6fe31bca9ab63f3c")
+    version("4.2.3", sha256="869ca4967cd255097fd2dc31664f30607e81f5abcf5f9c89bd467dc0bf93e5aa")
+    version("3.0.0", sha256="df2362351280edaf2233f3b2c8eb8e6dd6c68105f152897a4cc629fa346a7396")
+    version("2.2.2", sha256="6ffaad40505dbe8f155049448554b54ea31d31babf74ccf6b7935bfe55eeafd8")
+    version("2.2.1", sha256="c8577ba1ddd5c28fd0aa7c579ae65ab990eb7cb51ecf9f8175f9251f6deb9a06")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libxc/package.py:35]: It seems we now depend on the auto-generated archives. Can the shasum change again, due to e.g. Gitlab updates? Can't we depend on verified release tarballs instead?

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 2]
```diff
@@ -41,11 +41,10 @@ class Libxc(AutotoolsPackage, CudaPackage):
     conflicts("+shared +cuda", msg="Only ~shared supported with +cuda")
     conflicts("+cuda", when="@:4", msg="CUDA support only in libxc 5.0.0 and above")
 
-    # Remove this when the release tarballs become available for 6.0.0 and above.
-    with when("@6.0.0:"):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libxc/package.py:45]: Why was this removed?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 3]
```diff
@@ -57,17 +56,6 @@ class Libxc(AutotoolsPackage, CudaPackage):
         when="@6.0.0",
     )
 
-    patch("nvhpc-configure.patch", when="%nvhpc")
-    patch("nvhpc-libtool.patch", when="@develop %nvhpc")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libxc/package.py:61]: Similarly, are these no longer needed?

**Comment by RemiLacroix-IDRIS** [comment]
@alalazo: I should have updated the description of this PR with the content of my new commit message:
>  libxc: Fix compilation after distribution changes.
>
> The release tarballs are not available anymore which means autoconf, automake and libtool are always needed.
>
> The NVHPC specific patches don't make sense anymore being that the patched files are not distributed in the new tar files.

File: var/spack/repos/builtin/packages/libxc/package.py (hunk 1 above)
Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/libxc/package.py:35]: I don't know if the shasum can change due to Gitlab updates but libxc stopped distributing release tarballs.

Comment (alalazo) [reply var/spack/repos/builtin/packages/libxc/package.py:35]: :sob:

**Review by alalazo — APPROVED** [review]
